tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -4700,3 +4700,271 @@ def test_18_barbican(self):
+                ssh.close()
+
+                if md5sums_before_full == \
+                        md5sums_after_full_inplace:
+                    LOG.debug("***MDSUMS MATCH***")
+                    reporting.add_test_step(
+                        "File present on inplace restored instance", tvaultconf.PASS)
+                    tests[3][1] = 1
+                else:
+                    LOG.error("***MDSUMS DON'T MATCH***")
+                    raise Exception("Inplace restore of full snapshot - checksum failed.")
+            else:
+                LOG.error("Inplace restore of full snapshot failed. Status not available.")
+                raise Exception("Inplace restore of full snapshot")
+
+
+            reporting.test_case_to_write()
+
+
+            #oneclick restore
+            reporting.add_test_script(tests[4][0])
+
+            #delete original instance of vm.
+            self.delete_vm(self.vm_id)
+
+            time.sleep(10)
+            rest_details = {}
+            rest_details['rest_type'] = 'oneclick'
+            payload = self.create_restore_json(rest_details)
+            # Trigger oneclick restore of full snapshot
+            restore_id_3 = self.snapshot_inplace_restore(
+                self.wid, self.snapshot_id, payload)
+            self.wait_for_snapshot_tobe_available(self.wid, self.snapshot_id)
+            if(self.getRestoreStatus(self.wid, self.snapshot_id,
+                    restore_id_3) == "available"):
+
+                reporting.add_test_step("Oneclick restore of full snapshot",
+                        tvaultconf.PASS)
+
+                ssh = self.SshRemoteMachineConnectionWithRSAKey(fip[0])
+                md5sums_after_full_oneclick = self.calculatemmd5checksum(ssh, "/opt")
+                LOG.debug(f"md5sums_after_full_oneclick: {md5sums_after_full_oneclick}")
+                ssh.close()
+
+                if md5sums_before_full == \
+                        md5sums_after_full_oneclick:
+                    LOG.debug("***MDSUMS MATCH***")
+                    reporting.add_test_step(
+                        "File present on oneclick restored instance", tvaultconf.PASS)
+                    tests[4][1] = 1
+                else:
+                    LOG.error("***MDSUMS DON'T MATCH***")
+                    raise Exception("Oneclick restore of full snapshot - checksum failed")
+            else:
+                LOG.error("Oneclick restore of full snapshot failed. Status not available.")
+                raise Exception("Oneclick restore of full snapshot")
+
+
+            reporting.test_case_to_write()
+
+        except Exception as e:
+            LOG.error("Exception: " + str(e))
+            reporting.add_test_step(str(e), tvaultconf.FAIL)
+            reporting.set_test_script_status(tvaultconf.FAIL)
+
+        finally:
+            #delete secret container created.
+            self.delete_secret_container(self.container_ref)
```

Comment:
This need to be added to base.py as default cleanup call and not to be added here explicitly. You can refer to create_secret() method and have similar implementation.